### DISCUSSION TITLE
Should I use a single render pass for each target buffer?

### DISCUSSION THREAD

**Discussion by m-schuetz** [gpuweb/gpuweb#1698] [Q&A] [answered]
[WebGPU Spec on render pass load values](https://gpuweb.github.io/gpuweb/#enumdef-gpuloadop): 

> "Note: On some GPU hardware (primarily mobile), providing a clear-value is significantly cheaper because it avoids loading data from main memory into tile-local memory. On other GPU hardware, there isn’t a significant difference. As a result, it is recommended to use a clear-value, rather than "load"."

Does this mean that it is actually expensive to use multiple render passes to draw a single framebuffer? Using ```loadValue: "load"``` would simplify some render engine code - the thinking was to split render code paths into multiple independent modules, each of which starts a render pass into framebuffers that may or may not be the same. The first module to draw into a framebuffer would use "clear", all subsequent ones would use "load". 

I was hoping that "load" would allow me to simply continue working on a framebuffer without too much overhead. Kind of like how the [reverse-z](http://austin-eng.com/webgpu-samples/samples/reversedZ) demo uses "load" to continue drawing into the same framebuffer. But if "load" actually incurrs a copy of the whole framebuffer, I probably shouldn't do that.

**Comment by kvark** [comment, marked as answer]
"Load" is fine on discreet GPUs with their own memory, but it's slow on mobile GPUs. If you can have a single render pass, that's the best.

**Reply by m-schuetz** [reply]
Thanks, that's good to know!
